rust/feature-flags/src/api/endpoint.rs

```diff
@@ -41,22 +69,37 @@ pub async fn flags(
         body,
     };
 
-    let version = context
+    // Check if this request came through the decide proxy
+    let is_from_decide = headers
+        .get("X-Original-Endpoint")
```

Comment:
If we got this, but it's not from decide, would we want to log it because we made up this header so any value _other than_ `decide` would be suspect.

Perhaps not important in this PR.